### DISCUSSION TITLE
Add small time drift to podcast pubDate to correct playback order

### DISCUSSION THREAD

**Discussion by Jumbs** [advplyr/audiobookshelf#888] [Ideas]
Some podcast clients use the pubDate to sort. 
For audiobooks, that seems to be tied to the date added of the files, which is all the same.
This causes sorting to fail, and often have chapters/files out of order.

Adding a small time drift when generating the RSS XML pubDate. 
Time +/- Tracknumber might work.

**Comment by advplyr** [comment]
I fixed this for the next release. The pub date was being set for every audiobook "episode" as the library item added at date.
I updated it so that it adds a second for each audio track to make sure they are in the correct order.
https://github.com/advplyr/audiobookshelf/commit/1bfa4b31f278b743a6348a6b4713ee743dbe7477

**Reply by Jumbs** [reply]
Great, thanks!

**Reply by advplyr** [reply]
Fixed in [v2.1.2](https://github.com/advplyr/audiobookshelf/releases/tag/v2.1.2)

**Comment by hcbille** [comment]
To add to this, would it be possible to have setting to make it the reverse order. My Podcast app defaults to latest episode first, so whenever I add an audiobook  I need to change the settings specifically for that book each time. 

An setting that would reverse the order would be awesome! :D

**Comment by Jumbs** [comment]
see #952 and #1253, as it turned out to not be that simple. 
As is, this works on google podcasts and apple podcasts. I cannot speak to any other.
On Apple, even if you reverse the order visually, it would still play based on date.
